system/src/system_network_manager.cpp

```diff
@@ -609,11 +616,25 @@ void NetworkManager::handleIfLink(if_t iface, const struct if_event* ev) {
         } else if (state_ == State::IP_CONFIGURED || state_ == State::IFACE_LINK_UP) {
             refreshIpState();
         }
+
+        // If the preferred network becomes available, close the cloud connection to force migration to this network
+        if (ConnectionManager::instance()->getPreferredNetwork() == netIfIndex) {
```

Comment:
This works, but technically is not the correct spot to check if a network is ready (ie IP configured).
This state is just link up. The implicit assumption here is that the interface will be assigned an IP by the time we try to reopen the cloud socket, which usually is true.

I think to be strictly correct, the transition from link up -> ip configured is detected in `refreshIpState()`, so perhaps we should check for connection migration there?